### PR TITLE
Plot: ellipseRect, ellipseRectAA, lineWidth, setPixelAA

### PR DESCRIPTION
**Pull request by azrafe7**
This PR fixes some minor things in the methods listed.
I'm still not sure about setPixelAA, but that could be revised later.

### CODE REVIEW

**Comment by azrafe7** [comment]
It seems that you ported the code from the cpp version, but I just noticed that the js one is more recent and also contains more functions (AA and width versions of ~all methods, as well as some fixes). How to proceed... should we start over with the js one, or just keep going and integrate the missing functions while testing for correct output?

**Comment by Justinfront** [comment]
I looked at the js version as well ( sorry did not mention that ) and it helped me get the c++ version working so they are partially a mix.  Both C++ and js seem pretty sloppy languages compared to Haxe and the code is written very tight when in the port it's clearer. 
I don't rem seeing additional methods but maybe I missed some.  I did use it partially to work out how to port the code. I think it's best to stick with improving incrementally what we have it takes quite a lot of work to get the compiler not to complain I probably worked on it over a couple of days!  Actually after porting the code to the extent I did I came to the conclusion many of the for loops were overly complex and hard to follow it's obviously the work of a very smart c++ coder but I am sure some could be simpler.  Really we should work out what is useful and maybe remove the rest or comment it as not used or move it to a gist, a lot of it is overkill for a clean and simple graphics interface.  Drawing the curves with lineTo like I did for luxe is possibly a simpler cleaner solution than the version provided but maybe your getting better results now you have fixed them?  I think you will loose huge amout of time starting from scratch and I actually don't think we need so much methods.

**Comment by azrafe7** [comment]
I don't think you missed some cpp methods in the port, but the js added a `AA` and `width` version of nearly all methods. Anyway I agree with integrating the code by looking at the js ver.

Some innocent questions:
 - What you think should be left out, which methods?
 - What was the problem with luxe, cannot recall it?

**Comment by azrafe7** [comment]
Also... we'll probably need a demo like the original one at http://members.chello.at/~easyfilter/canvas.html to be sure things are working.

**Comment by Justinfront** [comment]
there is a copyright on the actual demo itself so 'unlike' would probably be more suitable :)
